ui/src/shared/utils/downloadTimeseriesCSV.ts

```diff
@@ -0,0 +1,100 @@
+import moment from 'moment'
+import {unparse} from 'papaparse'
+
+import {proxy} from 'src/utils/queryUrlGenerator'
+import {duration} from 'src/shared/apis/query'
+import replaceTemplates, {replaceInterval} from 'src/tempVars/utils/replace'
+import {timeSeriesToTableGraph} from 'src/utils/timeSeriesTransformers'
+
+import {Query, Template} from 'src/types'
+import {TimeSeriesServerResponse} from 'src/types/series'
+
+type CSV = Array<Array<number | string>>
+
+const downloadTimeseriesCSV = async (
+  queries: Query[],
+  templates: Template[]
+): Promise<void> => {
+  const responses = await Promise.all(
+    queries.map(query => fetchTimeseries(query, templates))
+  )
+
+  const csv = await timeseriesToCSV(responses)
+
+  downloadCSV(csv, csvName(csv))
+}
+
+const fetchTimeseries = async (
+  query: Query,
+  templates: Template[]
+): Promise<TimeSeriesServerResponse> => {
+  const source = query.queryConfig.source
+
+  let queryString = replaceTemplates(query.text, templates)
+
+  if (queryString.includes(':interval:')) {
+    const queryDuration = await duration(query.text, source)
+
+    queryString = replaceInterval(query.text, null, queryDuration)
+  }
+
+  const resp = await proxy({source: source.links.proxy, query: queryString})
+
+  return {response: resp.data}
+}
+
+const timeseriesToCSV = async (
+  responses: TimeSeriesServerResponse[]
+): Promise<CSV> => {
+  const tableResponse = await timeSeriesToTableGraph(responses)
+  const table = tableResponse.data
+
+  if (!table.length) {
+    throw new Error('no results')
+  }
+
+  const header = table[0]
+  const timeIndex = header.indexOf('time')
+
+  if (timeIndex < 0) {
+    throw new Error('could not find time index')
+  }
+
+  for (let i = 1; i < table.length; i++) {
+    // Convert times to a (somewhat) human readable ISO8601 string
+    table[i][timeIndex] = new Date(table[i][timeIndex]).toISOString()
+  }
+
+  return table
+}
+
+const csvName = (csv: CSV) => {
+  // Strange time format since `:` can't be used in a filename
+  const now = moment().format('YYYY-MM-DD-HH-mm')
+
+  let columns = []
+
+  if (csv.length) {
+    columns = csv[0]
+      .filter(col => col !== 'time')
+      .map((col: string) => col.replace('.', '_'))
+  }
+
+  return `${now} ${columns.join(' ')}.csv`
+}
+
+const downloadCSV = (csv: CSV, title: string) => {
+  const text = unparse(csv)
```

Comment:
I thought it would be nice to remove `download.js` soon, as it's kinda scary: https://github.com/influxdata/chronograf/blob/ceo-csv-export/ui/src/external/download.js

I can remove the existing usages of `download.js` in this PR if that seems good to you.

Reply:
oh wow i see. Yea then this looks fine. I dont think you need to remove the other uses of download.js in this PR. seems like that could be a separate pr to me.